fs/remote/resolver.go

```diff
@@ -303,6 +310,56 @@ func redirect(ctx context.Context, blobURL string, tr http.RoundTripper, timeout
 	return
 }
 
+func getLayerSize(ctx context.Context, hf *httpFetcher) (int64, error) {
+	if hf.timeout > 0 {
+		var cancel context.CancelFunc
+		ctx, cancel = context.WithTimeout(ctx, hf.timeout)
+		defer cancel()
+	}
+	req, err := http.NewRequestWithContext(ctx, "HEAD", hf.url, nil)
+	if err != nil {
+		return 0, err
+	}
+	req.Close = false
+	res, err := hf.tr.RoundTrip(req)
+	if err != nil {
+		return 0, err
+	}
+	defer res.Body.Close()
+	if res.StatusCode == http.StatusOK {
+		return strconv.ParseInt(res.Header.Get("Content-Length"), 10, 64)
+	}
+	headStatusCode := res.StatusCode
+
+	// Failed to do HEAD request. Fall back to GET.
+	// ghcr.io (https://github-production-container-registry.s3.amazonaws.com) doesn't allow
+	// HEAD request (2020).
+	req, err = http.NewRequestWithContext(ctx, "GET", hf.url, nil)
+	if err != nil {
+		return 0, errors.Wrapf(err, "failed to make request to the registry")
+	}
+	req.Close = false
+	req.Header.Set("Range", "bytes=0-1")
+	res, err = hf.tr.RoundTrip(req)
+	if err != nil {
+		return 0, errors.Wrapf(err, "failed to request")
```

Comment:
We seem to have migrated away from `errors.Wrap` in favor of `fmt.Errorf("%w")` since this code was written and removed. I will update to more recent conventions unless someone has input on why this should remain different.

Reply:
`errors.Wrapf` was from before `fmt.Errorf` was part of the standard library. We should use the more recent convention.